### PR TITLE
LOOP-1361 Capture manually entered BG reading

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-1361
- Use TSMBGDatum type if glucose is user entered

Requesting any 2 reviewers, please.

### CODE REVIEW

**Review by rickpasetto — APPROVED** [review]
So, I'm not sure what a `TSMBGDatum` is, but seems legit.
👍

**Comment by darinkrauss** [comment]
SMBGDatum stands for "Self Monitor Blood Glucose Datum" (i.e. user manual) and CBGDatum stands for "Continuous Blood Glucose Datum" (i.e. device). This is per the Tidepool Data Model (backend).

File: TidepoolServiceKit/Extensions/StoredGlucoseSample.swift [hunk 1]
```diff
@@ -13,6 +13,8 @@ extension StoredGlucoseSample {
     var datum: TDatum {
         if isDisplayOnly {
             return TCalibrationDeviceEventDatum(time: datumTime, value: datumValue, units: datumUnits).adorn(withOrigin: datumOrigin)
+        } else if wasUserEntered {
+            return TSMBGDatum(time: datumTime, value: datumValue, units: datumUnits, subType: .manual).adorn(withOrigin: datumOrigin)
```
Comment (nhamming) [inline TidepoolServiceKit/Extensions/StoredGlucoseSample.swift:17]: hm... will this work? Often `isDisplayOnly` is set when `wasUserEntered` is set, unless I missed something in the previous PRs. I mean, I have yet to see a case where only `wasUserEntered` is set.

**Review by nhamming — COMMENTED** [review]
Concern about the need for this code (but I'm probably lacking scope).

File: TidepoolServiceKit/Extensions/StoredGlucoseSample.swift (hunk 1 above)
Comment (darinkrauss) [reply TidepoolServiceKit/Extensions/StoredGlucoseSample.swift:17]: Loop will soon support manually entered glucose values for use during a bolus when CGM readings are not available for some reason. This glucose value will be non-calibration (`isDisplayOnly=false`) user-entered (`wasUserEntered=true`). This is the internal implementation and the UI will be implemented with https://tidepool.atlassian.net/browse/LOOP-906.

**Review by ps2 — APPROVED** [review]
LGTM

**Pull request merged**